OBSDOCS-1723: Port the "Installing Logging" sections to 5.8 and 6.y docs (#89053)
openshift/openshift-docs: modules/log6x-logging-clo-cli-install.adoc

// Module is included in the following assemblies:
//
// 
:_mod-docs-content-type: PROCEDURE
[id="log6x-installing-logging-operator_{context}"]
= Installing {clo} by using the CLI

> Install {clo} on your {product-title} cluster to collect and forward logs to a log store by using the {product-title} command-line interface (CLI).

Reviewer: ```suggestion
Install {clo} on your {product-title} cluster to collect and forward logs to a log store by using the {oc-first}.
```

> Install {clo} on your {product-title} cluster to collect and forward logs to a log store by using the  {product-title} command-line interface (CLI).

Reviewer 2: ```suggestion
Install {clo} on your {product-title} cluster to collect and forward logs to a log store by using the {product-title} command-line interface (CLI).
```

<4> Specify `redhat-operators` as the value. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object that you created when you configured Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

. Create a service account to be used by the log collector:
+
[source,terminal]
----
$ oc create sa logging-collector -n openshift-logging
----

. Assign the necessary permissions to the service account for the collector to be able to collect and forward logs. In this example, the collector is provided permissions to collect logs from both infrastructure and application logs.
+
[source,terminal]
----
$ oc adm policy add-cluster-role-to-user logging-collector-logs-writer -z logging-collector -n openshift-logging
$ oc adm policy add-cluster-role-to-user collect-application-logs -z logging-collector -n openshift-logging

> $ oc adm policy add-cluster-role-to-user collect-infrastructure-logs -z logging-collector -n openshift-logging

Reviewer: We do not include multiple commands in a single code block. Suggest to divide it in a separate instruction+command format.

Author: Considering the timeline for the release would you min if i do this as a part of https://issues.redhat.com/browse/OBSDOCS-1747?

.Example `ClusterLogForwarder` CR
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
  name: instance
  namespace: openshift-logging # <1>
spec:
  serviceAccount:
    name: logging-collector # <2>
  outputs:
  - name: lokistack-out
    type: lokiStack # <3>
    lokiStack:
      target: # <4>
        name: logging-loki 
        namespace: openshift-logging
      authentication:
        token:
          from: serviceAccount
    tls:
      ca:
        key: service-ca.crt
        configMapName: openshift-service-ca.crt
  pipelines:
  - name: infra-app-logs
    inputRefs: # <5>
    - application
    - infrastructure
    outputRefs:
    - lokistack-out
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify the name of the service account created before.
<3> Select the `lokiStack` output type to send logs to the `LokiStack` instance.
<4> Point the `ClusterLogForwarder` to the `LokiStack` instance created before.
<5> Select the log output types you want to send to the `LokiStack` instance.

. Apply the `ClusterLogForwarder CR` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> 

Reviewer 3: ```suggestion
.Verification
```

. Verify the installation by running the following command:
+
[source,terminal]
----
$ oc get pods -n openshift-logging
----
+
.Example output
[source,terminal]
----
$ oc get pods -n openshift-logging
NAME                                               READY   STATUS    RESTARTS   AGE
cluster-logging-operator-fb7f7cf69-8jsbq           1/1     Running   0          98m
collector-222js                                    2/2     Running   0          18m
collector-g9ddv                                    2/2     Running   0          18m
collector-hfqq8                                    2/2     Running   0          18m
collector-sphwg                                    2/2     Running   0          18m
collector-vv7zn                                    2/2     Running   0          18m

> collector-wk5zz                                    2/2     Running   0          18m

Reviewer 4: Since the name of ClusterLogForwarder is `instance`, the collector pods are named as `instance-xxxxx`.

Author: thanks, updated

.Procedure

. Create a `Subscription` object for {clo}:
+
.Example `Subscription` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: cluster-logging
  namespace: openshift-logging # <1>
spec:
  channel: stable-6.<y> # <2>
  installPlanApproval: Automatic # <3>
  name: cluster-logging
  source: redhat-operators # <4>
  sourceNamespace: openshift-marketplace
----
<1> You must specify the `openshift-logging` namespace.
<2> Specify `stable-6.<y>` as the channel.
<3> If the approval strategy in the subscription is set to `Automatic`, the update process initiates as soon as a new Operator version is available in the selected channel. If the approval strategy is set to `Manual`, you must manually approve pending updates.
<4> Specify `redhat-operators` as the value. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the `CatalogSource` object that you created when you configured the Operator Lifecycle Manager (OLM).

. Apply the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> . Create an `OperatorGroup` object:

Reviewer 4: Generally, we should create OperatorGroup at first, then create the Subscription. Or the subscription may raise error.

Author: updated here and in loki CLI as well.

+
.Example `OperatorGroup` object
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  annotations:

>     olm.providedAPIs: ClusterLogForwarder.v1.observability.openshift.io,LogFileMetricExporter.v1alpha1.logging.openshift.io

Reviewer 4: No need to add the annotations in the yaml file.

Reviewer 3: Not adding the annotation will not break anything on the UI? I asked @theashiot to add the annotation since it showed when installing the operator using the UI

Reviewer 4: No. The annotation is added automatically.

Author: removed
